scripts/send_hex.py

```diff
@@ -94,6 +94,29 @@ def chunkGenerator(data):
         j += MAX_HEX_CHUNK_BYTES
         yield data[i:j]
     yield data[i + MAX_HEX_CHUNK_BYTES:len(data)]
+
+
+def printData(ser):
+    """
+    Prints data sent to the computer from the Arduino over seral.
+    Transmissions are seperated by the HANDSHAKE_CHAR.
+    
+    Params:
+        ser:
+            Our initiallized pyserial serial port
+
+    Outputs:
+        None
+    """
+    data = ""
+    curr = ""
+    time.sleep(1) # wait for the Arduino
+    while curr !=  HANDSHAKE_CHAR:
+        while curr !=  HANDSHAKE_CHAR and ser.in_waiting:
+            data += curr
```

Comment:
Why do there need to be two while loops here to check if you have received the handshake character?